README.md

```diff
@@ -17,4 +17,108 @@ These expectations are as follows:
+
+It can also be manipulated programmatically, by manually or semi-automatically modifying the `:kind`, `:channel`, or other columns to correct for missed signals by the default labels (for which `:kind` and `:channel` will be `missing`).
+We give two examples of how such a workflow might work here: one where the plan is modified before being executed, and another where EDF signal headers are be _preprocessed_ before the plan is constructed.
+
+### Modification of a plan
+
+For instance, some EEG datasets have the physical units set to millivolts, but the signals are usually better measured in microvolts.
+During import, you want to correct this by adjusting the encoding settings used by Onda to store samples, by scaling the sample offset and resolution by 1000 and setting the physical units.
+This can be accomplished by modifying the rows of the plan like so:
+
+```julia
+edf = EDF.File(my_edf_file_path)
+plans = plan_edf_to_onda_samples(edf; label=my_labels)
+
+function fix_millivolts(plan)
+    if plan.sample_unit == "millivolt" && plan.kind == "eeg"
+        sample_resolution_in_unit = plan.sample_resolution_in_unit * 1000
+        sample_offset_in_unit = plan.sample_offset_in_unit * 1000
+        return Tables.rowmerge(plan; sample_unit="microvolt",
+                               sample_resolution_in_unit,
+                               sample_offset_in_unit)
+    else
+        return plan
+    end
+end
+
+new_plan = map(fix_millivolts, Tables.rows(plans))
+samples, plan_executed = edf_to_onda_samples(edf, new_plan)
+```
+
+As another, similar example, sometimes EMG channels get recorded with different physical units.
+In such a case, OndaEDF will store them with different `kind` values (`emg_1`, `emg_2`, etc.).
+This can be corrected in a similar way, for exmaple by converting millivolts to microvolts (adjusting of course depending on the nature of your dataset) and re-grouping into Onda signals:
+
+```julia
+edf = EDF.File(my_edf_file_path)
+plans = plan_edf_to_onda_samples(edf; label=my_labels)
+
+function fix_emg(plan)
+    if startswith(plan.kind, "emg")
+        if plan.sample_unit == "millivolt"
+            sample_resolution_in_unit = plan.sample_resolution_in_unit * 1000
+            sample_offset_in_unit = plan.sample_offset_in_unit * 1000
+            plan = Tables.rowmerge(plan; sample_unit="microvolt",
+                                   sample_resolution_in_unit,
+                                   sample_offset_in_unit)
+        end
+        return Tables.rowmerge(plan; kind="emg")
+    else
+        return plan
+    end
+end
+
+new_plan = map(fix_emg, Tables.rows(plans))
+# re-compute the grouping of EDF signals into Onda signals:
+new_plan = OndaEDF.plan_edf_to_onda_signal_groups(new_plan)
+samples, plan_executed = edf_to_onda_samples(edf, new_plan)
+```
```

Comment:
this scenario shows a more complicated post-processing where you try to merge emg_1 and emg_2 etc., which requires re-grouping